sources/lib/datalab/gcp/datalab/_bigquery.py

```diff
@@ -184,32 +183,37 @@ def _create_load_subparser(parser):
   return load_parser
 
 
-def _get_query_argument(args, config, env):
+def _get_query_argument(args, cell, env):
   """ Get a query argument to a cell magic.
 
   The query is specified with args['query']. We look that up and if it is a BQ query
   just return it. If it is instead a SqlModule or SqlStatement it may have variable
   references. We resolve those using the arg parser for the SqlModule, then override
-  the resulting defaults with either the Python code in config, or the dictionary in
+  the resulting defaults with either the Python code in cell, or the dictionary in
   overrides. The latter is for if the overrides are specified with YAML or JSON and
   eventually we should eliminate code in favor of this.
 
   Args:
     args: the dictionary of magic arguments.
-    config: the cell contents which can be variable value overrides.
+    cell: the cell contents which can be variable value overrides (if args has a 'query'
+        value) or inline SQL otherwise.
     env: a dictionary that is used for looking up variable values.
   Returns:
     A Query object.
   """
-  sql_arg = args['query']
+  sql_arg = args.get('query', None)
+  if sql_arg is None:
```

Comment:
if sql_arg is None, check and make sure cell is string. Otherwise, make sure cell is dict. This way we can print better error messages (instead of an exception thrown from env.update() or gcp.bigquery.Query() call)?